WordPressAuthenticator/GoogleSignIn/NewGoogleAuthenticator.swift

```diff
@@ -0,0 +1,97 @@
+import AuthenticationServices
+
+public class NewGoogleAuthenticator: NSObject {
```

Comment:
I created a new object to keep things tidy and made it `public` to use it in the demo app.

However, this code could, and probably should, be folded into `GoogleAuthenticator`.

I like the idea of exposing the building blocks for authentication in the library, so that consumers can decide whether to use the UI the library offers, or roll their own. But one could argue that's a bigger design change in the library and it ought to be considered and implemented outside the scope of this PR.

Reply:
> But one could argue that's a bigger design change in the library and it ought to be considered and implemented outside the scope of the SDK-less Google SignIn work.

I don't mind if this is a special case tbh – there are very very few consumers of this library, so I think to say "third party auth providers are different" is reasonable – especially since SIWA is like one button and a callback 😂